### PR TITLE
Add phpstan and resolve errors

### PR DESCRIPTION
**Pull request by hosni**

### CODE REVIEW

File: src/Providers/AutoDiscoverProvider.php [hunk 1]
```diff
@@ -18,6 +18,9 @@ public function register()
         $this->app->singleton(Contracts\IAutoDiscovery::class, Repository::class);
     }
 
+    /**
+	 * @return class-string[]
```
Comment (yeganemehr) [inline src/Providers/AutoDiscoverProvider.php]: this method can return any string.

File: src/Providers/AutoDiscoveryProvider.php [hunk 2]
```diff
@@ -18,6 +18,9 @@ public function register()
         $this->app->singleton(Contracts\IAutoDiscovery::class, Repository::class);
     }
 
+    /**
+	 * @return class-string[]
```
Comment (yeganemehr) [inline src/Providers/AutoDiscoveryProvider.php]: this method can return any string.

File: src/Repository.php [hunk 3]
```diff
@@ -2,9 +2,19 @@
...
 
 use Laravel\Lumen\Application;
-use Jalno\Lumen\Contracts\IAutoDiscovery;
+use Jalno\Lumen\Contracts;
 
-class Repository implements IAutoDiscovery
+/**
+ * @phpstan-type ComposerFile array{
+ *  "name": string,
+ *  "version": string,
```
Comment (yeganemehr) [inline src/Repository.php:10]: Why do you care about `name`, `version` and `description`?

File: src/Repository.php [hunk 4]
```diff
@@ -18,18 +28,27 @@ public function register(): void
...
+            }
+            foreach ($directories as $packageName) {
                 if (in_array($namespace, ['.', '..'])) {
                     continue;
                 }
                 $composer = $this->getComposerFrom($this->app->basePath("vendor/" . $namespace . "/" . $packageName . "/composer.json"));
                 if (empty($composer)) {
                     continue;
                 }
+                /** @var ComposerFile $composer */
```
Comment (yeganemehr) [inline src/Repository.php]: `$composer` variable is not defined here.

File: src/Repository.php [hunk 5]
```diff
@@ -38,14 +57,30 @@ public function register(): void
...
 
+    /**
+     * @return ComposerFile|array<null>
+     */
     private function getComposerFrom(string $path): array
     {
-        return is_file($path) ? json_decode(file_get_contents($path), true) : [];
+        if (is_file($path)) {
+            /** @var string $content */
+            $content = file_get_contents($path);
```
Comment (yeganemehr) [inline src/Repository.php:69]: check the `false`

File: src/Repository.php [hunk 6]
```diff
@@ -38,14 +57,30 @@ public function register(): void
...
 
+    /**
+     * @param ComposerFile $composer
+	 * @return class-string<Contracts\IPackage>[]
+	 */
     private function findJalnoPackageFrom(array $composer): array
     {
-        if (!isset($composer["extra"]["jalno"]["package"]) or empty($composer["extra"]["jalno"]["package"])) {
+        if (
+            !isset($composer["extra"]["jalno"]) or
```
Comment (yeganemehr) [inline src/Repository.php:85]: why you checking `$composer["extra"]["jalno"]`?

File: src/Repository.php [hunk 7]
```diff
@@ -38,14 +57,30 @@ public function register(): void
...
+            /** @var string $content */
+            $content = file_get_contents($path);
+            return json_decode($content, true);
+        }
+        return [];
     }
 
+    /**
+     * @param ComposerFile $composer
+	 * @return class-string<Contracts\IPackage>[]
```
Comment (yeganemehr) [inline src/Repository.php]: mind indentation.

File: composer.json [hunk 8]
```diff
@@ -9,6 +10,16 @@
     "require": {
         "jalno/lumen": "^0.0.1"
     },
+    "require-dev": {
+        "nunomaduro/larastan": "^0.7.6",
```
Comment (yeganemehr) [inline composer.json]: Why you needed larastan?

**Review by yeganemehr — CHANGES_REQUESTED** [review]

File: src/Providers/AutoDiscoverProvider.php (hunk 1 above)
Comment (hosni) [reply src/Providers/AutoDiscoverProvider.php]: Ok, I change it.

File: src/Providers/AutoDiscoveryProvider.php (hunk 2 above)
Comment (hosni) [reply src/Providers/AutoDiscoveryProvider.php]: OK, I change it.

File: src/Repository.php (hunk 3 above)
Comment (hosni) [reply src/Repository.php:10]: I removed them.

File: src/Repository.php (hunk 5 above)
Comment (hosni) [reply src/Repository.php:69]: I check that.

File: src/Repository.php (hunk 7 above)
Comment (hosni) [reply src/Repository.php]: Sorry about that, I fix it.

File: composer.json (hunk 8 above)
Comment (hosni) [reply composer.json]: As our last conversation about `nunomaduro/larastan`, I think I should use it for all packages.\
I removed it and use `phpstan/phpstan` instead.

File: src/Repository.php (hunk 6 above)
Comment (hosni) [reply src/Repository.php:85]: Cuz `phpstan` warn about that and says the `$composer["extra"]["jalno"]` maybe null.

**Pull request merged**